tests/testthat/test-integrate_sces.R

```diff
@@ -0,0 +1,214 @@
+# generate testing data
+set.seed(1665)
+merged_sce <- sim_sce(n_cells = 300, n_genes = 100, n_empty = 0)
+batches <- c(rep("a", 100),
+             rep("b", 100),
+             rep("c", 100))
+barcodes <- rownames(colData(merged_sce))
+
+# set up colData rownames as batch-barcode
+new_rownames <- tibble::tibble(
+  batches = batches,
+  barcodes = barcodes,
+  ids = glue::glue("{batches}-{barcodes}")) %>%
+  dplyr::pull(ids)
+rownames(colData(merged_sce)) <- new_rownames
+
+# Add some info to colData:
+# Add "sample" to colData
+colData(merged_sce)$sample <- batches
+# Add in a "covariate" column for testing
+colData(merged_sce)$covariate <- sample(c("patient-a", "patient-b", "patient-c", "patient-d"),
```

Comment:
This is not the way I would expect a covariate to work in real data, so it threw me for a second, but I don't think it matters for the test. But maybe not naming it `patient`.

Reply:
> but I don't think it matters for the test.

I was all in on this! I can remove `patient`, it's not needed.